core/src/createTextMaskInputElement.js

```diff
@@ -17,6 +17,56 @@ export default function createTextMaskInputElement({
   onReject,
   keepCharPositions = false
 }) {
+  // Anything that we will need to keep between `update` calls, we will store in this `state` object.
+  const previousConformedValue = emptyString
+  let state = {previousConformedValue}
+
+  return {
+    state,
+    // `update` is called by framework components whenever they want to update the `value` of the input element.
+    // The caller can send a `rawValue` to be conformed and set on the input element. However, the default use-case
+    // is for this to be read from the `inputElement` directly.
+    update: function(rawValue = inputElement.value) {
+      // `selectionStart` indicates to us where the caret position is after the user has typed into the input
+      const {selectionStart: currentCaretPosition} = inputElement
+
+      const nextMask = getNextMask({
+        rawValue,
+        previousConformedValue: state.previousConformedValue,
+        guide,
+        pipe,
+        providedMask,
+        placeholderChar,
+        onAccept,
+        onReject,
+        keepCharPositions,
+        currentCaretPosition,
+      })
+
+      if (nextMask !== undefined) {
+        const {value = '', adjustedCaretPosition} = nextMask
+        inputElement.value = value // set the input value
+        safeSetSelection(inputElement, adjustedCaretPosition) // adjust caret position
+        state.previousConformedValue = value // store value for access for next time
+      }
+    }
+  }
+}
+
+export function getNextMask({
```

Comment:
I think a better name for this function might be `getConformedInputState` or something like that...